contracts/Template.sol

```diff
@@ -1,106 +1,129 @@
-/*
- * SPDX-License-Identitifer:    GPL-3.0-or-later
- *
- * This file requires contract dependencies which are licensed as
- * GPL-3.0-or-later, forcing it to also be licensed as such.
- *
- * This is the only file in your project that requires this license and
- * you are free to choose a different license for the rest of the project.
- */
-
 pragma solidity 0.4.24;
 
-import "@aragon/os/contracts/factory/DAOFactory.sol";
-import "@aragon/os/contracts/apm/Repo.sol";
-import "@aragon/os/contracts/lib/ens/ENS.sol";
-import "@aragon/os/contracts/lib/ens/PublicResolver.sol";
-import "@aragon/os/contracts/apm/APMNamehash.sol";
-
-import "@aragon/apps-voting/contracts/Voting.sol";
-import "@aragon/apps-token-manager/contracts/TokenManager.sol";
-import "@aragon/apps-shared-minime/contracts/MiniMeToken.sol";
+import "@aragon/templates-shared/contracts/BaseTemplate.sol";
 
 import "./CounterApp.sol";
 
+// Using CompanyTemplate: https://github.com/aragon/dao-templates/tree/master/templates/company
 
-contract TemplateBase is APMNamehash {
-    ENS public ens;
-    DAOFactory public fac;
-
-    event DeployInstance(address dao);
-    event InstalledApp(address appProxy, bytes32 appId);
 
-    constructor(DAOFactory _fac, ENS _ens) public {
-        ens = _ens;
+contract Template is BaseTemplate {
+    string constant private ERROR_MISSING_TOKEN_CACHE = "COMPANY_MISSING_TOKEN_CACHE";
+    string constant private ERROR_EMPTY_HOLDERS = "COMPANY_EMPTY_HOLDERS";
+    string constant private ERROR_BAD_HOLDERS_STAKES_LEN = "COMPANY_BAD_HOLDERS_STAKES_LEN";
+    string constant private ERROR_BAD_VOTE_SETTINGS = "COMPANY_BAD_VOTE_SETTINGS";
```

Comment:
We can use a more suitable keyword here for the error messages instead of `COMPANY`, don't know if this template has a proper name... otherwise, I'd go with simply `TEMPLATE`